hnn_core/viz.py

```diff
@@ -270,7 +270,10 @@ def plot_dipole(dpl, tmin=None, tmax=None, ax=None, layer='agg', decim=None,
 
     if isinstance(dpl, Dipole):
         dpl = [dpl]
-    elif average:
+    for this_dpl in dpl:
+        _validate_type(this_dpl, Dipole, 'dpl', 'Dipole, list of Dipole')
+
+    if average:
         dpl = dpl + [average_dipoles(dpl)]
```

Comment:
@jasmainak what if someone happens to pass in a numpy array of Dipole objects? This would be an easy mistake to make and would pass all of our checks, but would get weird on L277 when the averaging step assumes `dpl + [average_dioles(dpl)]` is an appending operation.